operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+    radius: f64,
+    stddev: f64,
+}
+
+impl InitializedDensityRasterization {
+    fn new(
+        source: Box<dyn InitializedVectorOperator>,
+        result_descriptor: RasterResultDescriptor,
+        tiling_specification: TilingSpecification,
+        radius: f64,
+        stddev: f64,
+    ) -> Result<Self, error::Error> {
+        ensure!(
+            radius > 0.,
+            error::InvalidOperatorSpec {
+                reason: "The radius for density rasterization must be strictly positive."
+                    .to_string()
+            }
+        );
+
+        Ok(InitializedDensityRasterization {
+            source,
+            result_descriptor,
+            tiling_specification,
+            radius,
+            stddev,
+        })
+    }
+}
+
+impl InitializedRasterOperator for InitializedDensityRasterization {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> util::Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::F64(
+            DensityRasterizationQueryProcessor {
+                input: self.source.query_processor()?,
+                tiling_specification: self.tiling_specification,
+                radius: self.radius,
+                stddev: self.stddev,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct GridRasterizationQueryProcessor {
+    input: TypedVectorQueryProcessor,
+    spatial_resolution: SpatialResolution,
+    grid_size_mode: GridSizeMode,
+    tiling_specification: TilingSpecification,
+    origin_coordinate: Coordinate2D,
+}
+
+#[async_trait]
+impl RasterQueryProcessor for GridRasterizationQueryProcessor {
+    type RasterType = f64;
+
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        let grid_resolution = match self.grid_size_mode {
+            GridSizeMode::Fixed => SpatialResolution {
+                x: f64::max(self.spatial_resolution.x, query.spatial_resolution.x),
+                y: f64::max(self.spatial_resolution.y, query.spatial_resolution.y),
+            },
+            GridSizeMode::Relative => SpatialResolution {
+                x: f64::max(
+                    self.spatial_resolution.x * query.spatial_resolution.x,
+                    query.spatial_resolution.x,
+                ),
+                y: f64::max(
+                    self.spatial_resolution.y * query.spatial_resolution.y,
+                    query.spatial_resolution.y,
+                ),
+            },
+        };
+
+        let tiling_strategy = self
+            .tiling_specification
+            .strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+        let tile_shape = tiling_strategy.tile_size_in_pixels;
+
+        let tiles = ctx.thread_pool().install(|| {
```

Comment:
Do you do anything parallel in here?

Reply:
Good catch, not anymore. These simple additions didn't really seem to benefit from it so I removed the parallel computations. https://github.com/geo-engine/geoengine/pull/683/commits/892b2e8fe4f0d3e3d20dad595b96005e9af51462

Reply:
What you should do, in any case, is wrap a computation by a `spawn_blocking`.
The reason is that async stuff relies on collaborative scheduling, which means returning after a very short time to let others also use the thread pool. This means everything inside async and between calls of `.await`, which releases scope, should be kind of quick. Thus, computations should be inside such blocks, which you need to await and do the computation inside another "thread".

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/fc5eea38ef9908e97d8958a7ef60d36d06618563
I now use the tokio mpsc channel and `spawn_blocking` and got rid of the rayon scope in the process.
I think the loop in line 301 should be fast enough to stay as is. If you disagree I can change that as well.